File: LeetCode/cpp/795.cpp
class Solution {
public:
    int numSubarrayBoundedMax(vector<int> &A, int L, int R) {
        int i = 0, j = 0, m = -1, total = 0;
        while (j < A.size()) {
            if (A[j] >= L && A[j] <= R) {
                m = j;
                total += j - i + 1;
            } else if (A[j] < L) {
                total += m != -1 ? m - i + 1 : 0;
            } else if (A[j] > R) {
                i = j + 1;
                m = -1;
            }
            ++j;
        }
        return total;
    }
};

File: LeetCode/python3/795.py
class Solution:
    def numSubarrayBoundedMax(self, A: 'List[int]', L: 'int', R: 'int') -> 'int':
        i, m, j, res = 0, -1, 0, 0
        while j < len(A):
            if A[j] >= L and A[j] <= R:
                m = j
                res += m - i + 1
            elif A[j] < L:
                res += m - i + 1 if m != -1 else 0
            else:
                i = j + 1
                m = -1
            j += 1
        return res